applications/GeoMechanicsApplication/custom_strategies/strategies/residualbased_newton_raphson_strategy_linear_elastic_dynamic.hpp

```diff
@@ -0,0 +1,503 @@
+    using BaseType = ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>;
+    using ClassType =
+        GeoMechanicNewtonRaphsonStrategyLinearElasticDynamic<TSparseSpace, TDenseSpace, TLinearSolver>;
+    using TBuilderAndSolverType    = typename BaseType::TBuilderAndSolverType;
+    using TDataType                = typename BaseType::TDataType;
+    using SparseSpaceType          = TSparseSpace;
+    using TSchemeType              = typename BaseType::TSchemeType;
+    using DofsArrayType            = typename BaseType::DofsArrayType;
+    using TSystemMatrixType        = typename BaseType::TSystemMatrixType;
+    using TSystemVectorType        = typename BaseType::TSystemVectorType;
+    using LocalSystemVectorType    = typename BaseType::LocalSystemVectorType;
+    using LocalSystemMatrixType    = typename BaseType::LocalSystemMatrixType;
+    using TSystemMatrixPointerType = typename BaseType::TSystemMatrixPointerType;
+    using TSystemVectorPointerType = typename BaseType::TSystemVectorPointerType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * Default constructor
+     * @param rModelPart The model part of the problem
+     * @param pScheme The integration scheme
+     * @param pNewLinearSolver The linear solver employed
+     * @param pNewConvergenceCriteria The convergence criteria employed
+     * @param pNewBuilderAndSolver The builder and solver
+     * @param rParameters The parameters for the strategy
+     * @param Beta The Newmark Beta parameter
+     * @param Gamma The Newmark Gamma parameter
+     * @param MaxIterations The maximum number of non-linear iterations to be considered when solving the problem
+     * @param CalculateReactions The flag for the reaction calculation
+     * @param MoveMeshFlag The flag that allows to move the mesh
+     */
+    explicit GeoMechanicNewtonRaphsonStrategyLinearElasticDynamic(
+        ModelPart&                                 rModelPart,
+        typename TSchemeType::Pointer              pScheme,
+        typename TLinearSolver::Pointer            pNewLinearSolver,
+        typename TConvergenceCriteriaType::Pointer pNewConvergenceCriteria,
+        typename TBuilderAndSolverType::Pointer    pNewBuilderAndSolver,
+        Parameters&                                rParameters,
+        int                                        MaxIterations      = 30,
+        bool                                       CalculateReactions = false,
+        bool                                       MoveMeshFlag       = false)
+        : ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(
+              rModelPart, pScheme, pNewConvergenceCriteria, pNewBuilderAndSolver, MaxIterations, CalculateReactions, false, MoveMeshFlag)
+    {
+        // new constructor
+    }
+
+    /**
+     * @brief Initialization of member variables and prior operations
+     */
+    void Initialize() override
+    {
+        KRATOS_TRY;
```

Comment:
Please remove the empty statement:
```suggestion
        KRATOS_TRY
```